### PR TITLE
mgr/cephadm: replace async_map_completion with a simple wrapper

### PR DESCRIPTION
**Pull request by sebastian-philipp**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/cephadm/module.py [hunk 1]
```diff
@@ -2010,8 +1960,9 @@ def zap_device(self, host, path):
             raise OrchestratorError('Zap failed: %s' % '\n'.join(out + err))
         return '\n'.join(out + err)
 
+    @trivial_completion
     def blink_device_light(self, ident_fault, on, locs):
-        @async_map_completion
+        @forall_hosts
```
Comment (liewegas) [inline src/pybind/mgr/cephadm/module.py]: this should run on just one host i think!

File: src/pybind/mgr/cephadm/module.py [hunk 2]
```diff
@@ -2263,7 +2214,7 @@ def _create_daemon(self, daemon_type, daemon_id, host,
         return "{} {} on host '{}'".format(
             'Reconfigured' if reconfig else 'Deployed', name, host)
 
-    @async_map_completion
+    @forall_hosts
```
Comment (liewegas) [inline src/pybind/mgr/cephadm/module.py]: same

File: src/pybind/mgr/cephadm/module.py [hunk 3]
```diff
@@ -2532,7 +2483,7 @@ def _create_daemons(self, daemon_type, spec, daemons,
             )
             daemons.append(sd)
 
-        @async_map_completion
+        @forall_hosts
```
Comment (liewegas) [inline src/pybind/mgr/cephadm/module.py]: same

File: src/pybind/mgr/cephadm/module.py (hunk 1 above)
Comment (sebastian-philipp) [reply src/pybind/mgr/cephadm/module.py]: yes. right now, it's executed for all locs in parallel

Comment (sebastian-philipp) [reply src/pybind/mgr/cephadm/module.py]: `forall_hosts` doesn't really work for all "hosts", but maps over all arguments in parallel. Maybe `s/forall_hosts/forall_args/`?

**Comment by liewegas** [comment]
```
cephadm/module.py: note: In function "forall_hosts_wrapper":
cephadm/module.py:485: error: Item "None" of "Optional[CephadmOrchestrator]" has no attribute "_worker_pool"
```

**Pull request merged**